oss-internship-2020/curl/examples/example2.cc

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Sandboxed version of getinmemory.c
+// HTTP GET request using callbacks
+
+#include <cstdlib>
+#include <iostream>
+
+#include "../sandbox.h"
+
+struct MemoryStruct {
+  char* memory;
+  size_t size;
+};
+
+int main(int argc, char* argv[]) {
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
+  google::InitGoogleLogging(argv[0]);
+
+  absl::Status status;
+  sapi::StatusOr<CURL*> status_or_curl;
+  sapi::StatusOr<int> status_or_int;
+
+  // Initialize sandbox2 and sapi
+  CurlSapiSandbox sandbox;
+  status = sandbox.Init();
+  if (!status.ok()) {
+    LOG(FATAL) << "Couldn't initialize Sandboxed API: " << status;
+  }
+  CurlApi api(&sandbox);
+
+  // Generate pointer to WriteMemoryCallback function
+  sapi::RPCChannel rpcc(sandbox.comms());
+  size_t (*_function_ptr)(void*, size_t, size_t, void*);
+  status = rpcc.Symbol("WriteMemoryCallback", (void**)&_function_ptr);
```

Comment:
Don't use C style casts
https://google.github.io/styleguide/cppguide.html#Casting
I would probably invert the conversion - fetch void* from Symbol and then convert that to the function pointer if needed.
However here it seems to be completely unnecessary just declare `function_ptr` as `void*`.